### DISCUSSION TITLE
What is the weight matrix for score calculation?

### DISCUSSION THREAD

**Discussion by yuz12012** [networkx/networkx#5320] [Q&A]
Hi all,

Just want a clarification for the weight matrix used in clustering coefficient. The weight should represent distance between the nodes or the connection strength? 

Betweenness and closeness centrality calculation seems to be based on distance matrix?

Many thanks in advance.

**Comment by dschult** [comment]
The clustering coefficient simply takes the geometric average of the edge weights. This is independent of whether a weight means "distance" or "connection strength".  It will handle both the same way -- there are no shortest paths or closeness measures involved in the clustering coefficient.

**Reply by yuz12012** [reply]
Sorry for the confusion. 

But usually for weighted graph we assign the edge weights in two ways. One is connection strength and the other way is node distance. 

Like the networkx closeness centrality and betweeness centrality calculation, it definitely uses the distance weight matrix as input. I am just wondering what would be a proper assumption for clustering coefficient. High distance nodes generating high clustering coefficient seems to be inappropriate.

**Reply by dschult** [reply]
The clustering coefficient does not have a notion of distance or connection strength. It represents the fraction of possible triangles that are present in the graph. If you use weighted edges, then the triangles are weighted using the geometric average of the edge weights. The fraction of triangle weights that are present in the graph is the clustering coefficient.
